misc/bstr.h

```diff
@@ -80,6 +80,10 @@ double bstrtod(struct bstr str, struct bstr *rest);
 void bstr_lower(struct bstr str);
 int bstr_sscanf(struct bstr str, const char *format, ...);
 
+// Decode a string containing hexadecimal data. All non-hex characters will
+// be silently ignored, as will a lone trailing digit.
+struct bstr bstr_decode_hex(void *talloc_ctx, struct bstr hex);
```

Comment:
I don't understand. Ignore non-whitespace only, but error on whitespace? Also, due to the nature of ignoring non-hex, I think it's both semantically and practically more valid to assume it consumes the *entire* input, rather than providing some mechanism of returning the “parts that we looked at but contained no more valid hex digits”. That API would also have no users currently.